### ISSUE TITLE
`cache=False` and `cache='disk'` take same time for running one epoch

### ISSUE THREAD

**Issue by tejas-lica** [ultralytics/ultralytics#22426]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

First, thank you for this amazing architecture and the repository.

I am training YOLOv11 on 8x H100 GPUs using this repository on a custom dataset which has little over 1M samples in training set.
I am using batch_size=832 ( occupies ~70 GB of GPU RAM).
I have tried training with cache=False and cache='disk' to compare the time it takes to complete one epoch.
Based on my observations, it took the same amount of time ( ~14 mins) to complete 1 training epoch ( observed for 100+ epochs).

As per the [training documentation](https://docs.ultralytics.com/modes/train/#train-settings), I was expecting significant speedup for cache='disk' case compared to cache=False case.
I am sharing my training script here for reference.
It would be great if anyone could help me understand/explain why I did not get any speedup?

Sharing training script for reference.
[train.py](https://github.com/user-attachments/files/22956382/train.py)

Mithril node specs:
```
Linux yolo-2a-1-1 6.8.0-71-generic #71-Ubuntu SMP PREEMPT_DYNAMIC Tue Jul 22 16:52:38 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux
RAM: 1.6Ti
disk: 11TB
cpu: Intel Xeon Processor (SapphireRapids)
gpu: 8x H100
```

Thanks!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tejas-lica, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For your specific question about cache behavior on YOLO11 with a very large dataset:
-Please share a minimal reproducible snippet (CLI command or minimal Python script) that reproduces identical epoch times for `cache=False` vs `cache='disk'`
-Include your exact `ultralytics` version and environment details (e.g., `pip show ultralytics` or `yolo -v`)
-Add your dataloader settings: `workers`, `pin_memory`, `persistent_workers`, `prefetch_factor` (if set), `batch` (global vs per-GPU), `imgsz`, and `device`
-Clarify storage and filesystem: local NVMe vs network storage (NFS/SMB/Obj), disk type, and whether images are compressed (JPEG/PNG) and their average size
-Share representative training logs showing per-epoch timing and any dataloader/GPU utilization notes
-Confirm DDP/MP launch details (e.g., 8 GPUs, launch method) and whether epoch time is dominated by dataloader or compute
-If you believe this is a bug, please include an MRE using a small public dataset (e.g., 1–2k images) demonstrating no speedup with `cache='disk'` vs `cache=False` and the exact command you ran

These details will help us assess whether behavior is I/O-bound, compute-bound, or configuration-related 📈

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?
